### PR TITLE
Add Civetweb README

### LINKED ISSUE #18212: README file missing for civetweb sample
README file missing in samples/net/sockets/civetweb/.

### PR DESCRIPTION
**Pull request by tgorochowik**
This PR adds a minimal README for the Civetweb sample as requested in: #18212

This fixes #18212

### CODE REVIEW

**Review by rveerama1 — APPROVED** [review]
LGTM

**Review by jukkar — APPROVED** [review]

File: samples/net/sockets/civetweb/README.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/sockets/civetweb
+   :board: sam_e70_xplained
+   :goals: build
+   :compact:
+
+The sample application uses a static IP configuration.
+
+After flashing the board, the sever can be accessed with the web browser of your choice at ``10.0.0.111:8080``.
```
Comment (dbkinder) [inline samples/net/sockets/civetweb/README.rst]: change ``sever`` to ``server``

File: samples/net/sockets/civetweb/README.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+The IP configuration can be changed in Zephyr config.
+The default port can be changed in the sources of the sample.
```
Comment (dbkinder) [inline samples/net/sockets/civetweb/README.rst]: You should mention that this sample is not a general HTTP server accessing a filesystem when receiving http requests.  Instead only three "uris" are supported:  ``/``, ``/info``, and ``/history``, and explain the uri responses and what happens if any non-supported uris are received.

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: samples/net/sockets/civetweb/README.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
+.. _sockets-civetweb-sample:
+
+Civetweb sample
+###############
+
+Overview
+********
+
+This sample is a demonstrator of the HTTP API provided by the Civetweb module.
```
[the remark this answers is not in the document]
Comment (tgorochowik) [reply samples/net/sockets/civetweb/README.rst]: Updated

File: samples/net/sockets/civetweb/README.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+It makes Zephyr act as a HTTP server and demonstrates selected features of the Civetweb API.
+
+The source code for this sample application can be found at:
+:zephyr_file:`samples/net/sockets/civetweb`.
+
+Requirements
+************
+
+- A board with hardware networking
```
[the remark this answers is not in the document]
Comment (tgorochowik) [reply samples/net/sockets/civetweb/README.rst]: Updated

File: samples/net/sockets/civetweb/README.rst (hunk 1 above)
Comment (tgorochowik) [reply samples/net/sockets/civetweb/README.rst]: Fixed

File: samples/net/sockets/civetweb/README.rst (hunk 2 above)
Comment (tgorochowik) [reply samples/net/sockets/civetweb/README.rst]: Updated

**Comment by tgorochowik** [comment]
Thank you for the review @dbkinder !
Updated according to your comments.

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Comment by ioannisg** [comment]
Goes into 2.0 as documentation fix

**Review by ioannisg — APPROVED** [review]

**Pull request merged**